pkg/controller/argocd/secret.go

```diff
@@ -379,6 +385,38 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
+	secret := argoutil.NewSecretWithSuffix(cr.ObjectMeta, "namespaces")
+	secret.Labels[common.ArgoCDClusterSecretLabel] = "cluster"
+	dataBytes, _ := json.Marshal(map[string]interface{}{
+		"tlsClientConfig": map[string]interface{}{
+			"insecure": false,
+		},
+	})
+
+	secret.Data = map[string][]byte{
+		"config":     dataBytes,
+		"name":       []byte("in-cluster"),
+		"server":     []byte(common.ArgoCDDefaultServer),
+		"namespaces": []byte(cr.Namespace),
+	}
+
+	applyReconcilerHook(cr, secret, "")
```

Comment:
What will this actually do?

Reply:
This reconciler hook would be applied if the ArgoCD instance is a cluster config instance(openshift specific). For those instances, `namespaces` would be empty, which would allow the instance to access cluster scoped resources.

Reply:
So we'll have a hook in GitOps Operator that sets `.data.namespaces` to the empty string?

Reply:
Yeah, basically it just deletes the key. 
https://github.com/argoproj-labs/argocd-operator/pull/318/files#diff-464eb8ec36b8f4f45d63589aa754c7933c5db014ec7185c15fa38198718d7303R63